src/multibody/geometry.hxx

```diff
@@ -191,18 +205,17 @@ namespace se3
     activeCollisionPairs[pairId] = false;
   }
 
-  inline CollisionResult GeometryData::computeCollision(const CollisionPair & pair) const
+  inline bool GeometryData::computeCollision(const CollisionPair & pair, fcl::CollisionResult& collisionResult) const
   {
     const Index & co1 = pair.first;     assert(co1<collisionObjects.size());
     const Index & co2 = pair.second;    assert(co2<collisionObjects.size());
 
-    fcl::CollisionRequest collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP);
-    fcl::CollisionResult collisionResult;
+    static const fcl::CollisionRequest collisionRequest (1, false, false, 1, false, true, fcl::GST_INDEP);
```

Comment:
You may want to benchmark that. My guess is that, since the object is only locally used in the function, there would be no gain in pre-allocating it in Data. I dont think neither that the keyword static is interesting here, and might be counterproductive if we parallelize the calling algo.